test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/ClassTest.scala

```diff
@@ -100,13 +112,26 @@ class ClassTest {
 
     assertTrue(classOf[Object].isAssignableFrom(classOf[String]))
     assertTrue(classOf[Seq[_]].isAssignableFrom(classOf[List[_]]))
+    assertTrue(classOf[List[_]].isAssignableFrom(classOf[::[_]]))
+    assertTrue(classOf[Seq[_]].isAssignableFrom(classOf[::[_]]))
     assertTrue(classOf[Object].isAssignableFrom(classOf[Array[String]]))
     assertTrue(classOf[Array[Seq[_]]].isAssignableFrom(classOf[Array[List[_]]]))
 
     assertFalse(classOf[String].isAssignableFrom(classOf[Object]))
     assertFalse(classOf[List[_]].isAssignableFrom(classOf[Seq[_]]))
+    assertFalse(classOf[Option[_]].isAssignableFrom(classOf[::[_]]))
+    assertFalse(classOf[Set[_]].isAssignableFrom(classOf[::[_]]))
```

Comment:
I do not understand what these two new cases are adding in terms of coverage.

Reply:
They're covering cases where the left hand side and/or right hand side is a *class* (not a trait) that is not special from the linker's perspective. Instance tests for classes use `instanceof` now, so it's good to have tests for that.